gnd/src/main/java/com/google/android/gnd/ui/editrecord/EditRecordViewModel.java

```diff
@@ -78,6 +84,56 @@
     this.showUnsavedChangesDialogEvents = new SingleLiveEvent<>();
     this.showErrorDialogEvents = new SingleLiveEvent<>();
     this.authManager = authenticationManager;
+    this.editRecordRequests = PublishSubject.create();
+    this.recordSaveRequests = PublishSubject.create();
+
+    disposeOnClear(
+        recordSaveRequests
+            .switchMap(this::saveRecord)
```

Comment:
Don't worry about it here, but we should change this to use the `doOnError` handling instead of handling the error in the subscription.